src/app/code/community/Omise/Gateway/Controller/Base.php

```diff
@@ -1,19 +1,153 @@
 <?php
 abstract class Omise_Gateway_Controller_Base extends Mage_Core_Controller_Front_Action
 {
+    /**
+     * @var string
+     */
+    protected $message;
+
+    /**
+     * @var string
+     */
+    protected $transId;
+
+    /**
+     * @var Omise_Gateway_Model_Order
+     */
+    protected $order;
+
+    /**
+     * @var string
+     */
+    protected $awaitingOrderStatus;
+
+    /**
+     * @return string
+     */
+    public function getMessage()
+    {
+        return $this->message;
+    }
+
+    /**
+     * @param string $message
+     */
+    public function setMessage($message)
+    {
+        $this->message = $message;
+    }
+
+    /**
+     * constructor
+     */
     protected function _construct()
     {
         $omise = Mage::getModel('omise_gateway/omise');
         $omise->initNecessaryConstant();
+        $this->awaitingOrderStatus = Mage_Sales_Model_Order::STATE_PAYMENT_REVIEW;
     }
 
     /**
-     * @return \Omise_Gateway_Model_Order
+     * @return Omise_Gateway_Model_Order
      */
     protected function _getOrder()
     {
         $id = $this->getRequest()->getParam('order_id') ? $this->getRequest()->getParam('order_id') : null;
```

Comment:
Probably better to use:
`$this->getRequest()->getParam('order_id') ?: null;`
to avoid calling all this stuff twice.